test/org/github/jamm/MemoryMeterTest.java

```diff
@@ -66,578 +32,114 @@ public MemoryMeterTest(MemoryMeter.Guess guess)
-
-        assertEquals("Shallow size of char[0] " + guess, charArraySize(0), meter.measure(new char[0]));
-
-        assertEquals("Shallow size of char[1] " + guess, charArraySize(1), meter.measure(new char[1]));
-
-        assertEquals("Shallow size of char[256] " + guess, charArraySize(256), meter.measure(new char[256]));
-    }
-
-    @Test
-    public void testIntArraySizes() {
-        MemoryMeter meter = MemoryMeter.builder().withGuessing(guess).build();
-
-        assertEquals("Shallow size of int[0] " + guess, intArraySize(0), meter.measure(new int[0]));
-
-        assertEquals("Shallow size of int[1] " + guess, intArraySize(1), meter.measure(new int[1]));
-
-        assertEquals("Shallow size of int[256] " + guess, intArraySize(256), meter.measure(new int[256]));
+        Recursive dummy = new Recursive();
+        long shallowSize = meter.measure(dummy);
+        assertEquals("Deep size of Recursive is shallow size when child==null", shallowSize, meter.measureDeep(dummy));
+        dummy.child = dummy;
+        assertEquals("Deep size of Recursive is shallow size when child==this", shallowSize, meter.measureDeep(dummy));
     }
 
     @Test
-    public void testLongArraySizes() {
+    public void testDeep() {
         MemoryMeter meter = MemoryMeter.builder().withGuessing(guess).build();
 
-        assertEquals("Shallow size of long[0] " + guess, longArraySize(0), meter.measure(new long[0]));
-
-        assertEquals("Shallow size of long[1] " + guess, longArraySize(1), meter.measure(new long[1]));
-
-        assertEquals("Shallow size of long[256] " + guess, longArraySize(256), meter.measure(new long[256]));
-    }
-
-    @SuppressWarnings("unused")
-    static class LongHolder {
-        private long value;
-    }
-
-    @SuppressWarnings("unused")
-    static class IntHolder {
-        private int value;
-    }
-
-    @SuppressWarnings("unused")
-    static class CharHolder {
-        private char value;
-    }
-
-    @SuppressWarnings("unused")
-    static class TwoCharHolder {
-        private char value;
-        private char other;
-    }
-
-    @SuppressWarnings("unused")
-    static class ThreeCharHolder {
-        private char value;
-        private char other;
-        private char overflow;
+        Recursive root = new Recursive();
+        Recursive recursive = root;
+        for (int i = 0; i < 100000; i++) {
+            recursive.child = new Recursive();
+            recursive = recursive.child;
+        }
+        long shallowSize = meter.measure(root);
+        assertEquals(shallowSize * 100001, meter.measureDeep(root));
     }
 
     @SuppressWarnings("unused")
-    static class IntCharHolder {
-        private int value;
-        private char other;
+    private static class Parent {
+        private int i;
     }
 
     @SuppressWarnings("unused")
-    static class LongIntHolder {
-        private long value;
-        private int other;
+    private static class Child extends Parent {
```

Comment:
Hmmm so I still see the Child and Parent classes but the `testInheritance` that was using those is not presented anymore? I assume now you test testHierarchyPadding with A, B, C and that is why